### PR TITLE
[Reader Improvements] Feed post default card new design

### PR DESCRIPTION
**Pull request by thomashorta**
Part of #19082 

This currently updates the post card used in our feeds inside the Reader. This only affects the "default" card type, and the other card types (from the task list in https://github.com/wordpress-mobile/WordPress-Android/issues/19082) will be done in separate PRs.

To test:
1. Install and log in to Jetpack app
2. Make sure the `ReaderImprovementsFeatureConfig` is disabled in debug settings
3. **Verify** the post card UI did not change compared to previous versions
4. Enable the `ReaderImprovementsFeatureConfig` in debug settings
5. **Verify** the post card UI matches the specs (zQnohyMpLzBzQ5jzMMKni3-fi-869%3A19834)

## Regression Notes
1. Potential unintended areas of impact
Behavior changes when FF is off and in other parts of the app (such as Post Details screen)

7. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit testing.

8. What automated tests I added (or what prevented me from doing so)
Unit tests for major logic changes.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Review by osullivanchris — APPROVED** [review]
this is looking good so far! I don't have any issues with this PR but will spend some time using it and chat to you.

File: WordPress/src/main/res/layout/reader_include_post_detail_footer_new.xml [hunk 1]
```diff
@@ -82,7 +82,7 @@
             android:layout_width="0dp"
             android:layout_height="wrap_content"
             android:text="@string/like"
-            android:contentDescription="@string/likes_on_my_comments"
```
Comment (develric) [inline WordPress/src/main/res/layout/reader_include_post_detail_footer_new.xml:85]: interesting find, not sure why it was pointing to `likes_on_my_comments` 🤔 but seems more consistent as you changed it 👍

File: WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt [hunk 2]
```diff
@@ -141,7 +143,6 @@ class ReaderPostUiStateBuilder @Inject constructor(
             excerpt = buildExcerpt(post),
             title = buildTitle(post),
             tagItems = buildTagItems(post, onTagItemClicked),
-            photoFrameVisibility = buildPhotoFrameVisibility(post),
```
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt:144]: It was not used anymore, good catch 👍

**Review by develric — APPROVED** [review]
Hey @thomashorta 👋 , thanks for the changes. I can tell it was not straightforward to cut multiple PRs from the changes but code LGTM 👍! Also, I tested the changes with feature flag on and off and didn't get deviations from expected. :shipit:

**Pull request merged**